earthfile2llb/interpreter.go

```diff
@@ -445,6 +445,10 @@ func (i *Interpreter) handleRun(ctx context.Context, cmd spec.Command) error {
 	// Note: Not expanding args for the run itself, as that will be take care of by the shell.
 
 	if i.local {
+		if i.target.IsRemote() {
+			return i.errorf(cmd.SourceLocation, "Permission denied: unwilling to run locally command from remote Earthfile")
+		}
```

Comment:
It might be better if this is placed in `handleLocally`. Then it'll cover any command, not just `RUN`.

Reply:
oh that's a spectacular point! thanks for catching that.